### PR TITLE
Update guidelines about creating a new repository

### PR DESCRIPTION
**Pull request by Ressetkk**
Updated new repository template and community guidelines that reflect new review and approval flow with Prow.

**Related issue(s)**

#642

### CODE REVIEW

File: docs/guidelines/repository-guidelines/01-new-repository-settings.md
Comment (klaudiagrz) [inline docs/guidelines/repository-guidelines/01-new-repository-settings.md:55]:
```suggestion
Additionally, you can [define repository exclusive labels](https://help.github.com/en/articles/creating-a-label) for the new repository so you could use them in issues and pull requests. Follow the naming convention and color array used in other repositories in Kyma, such as the [`kyma`](https://github.com/kyma-project/kyma/labels) repository.
```

Comment (Ressetkk) [reply docs/guidelines/repository-guidelines/01-new-repository-settings.md:55]: I meant labels that are valid only for given repository. I believe `repository specific labels` or just `repository labels` would be enough to describe that.

File: templates/repository-template/OWNERS
Comment (Ressetkk) [reply templates/repository-template/OWNERS:17]: Reviewers look for general code quality, correctness, sane software engineering, style, etc.
Approvers look for holistic acceptance criteria, including dependencies with other features, forwards/backwards compatibility, API and flag definitions, etc.
When a person is in both reviewers and approvers groups, their Approving review is also considered as an approve and marks the PR as `approved`. That behaviour skips the requirement for a second review and immediately allows the PR to be merged.

File: docs/guidelines/repository-guidelines/01-new-repository-settings.md
Comment (klaudiagrz) [inline docs/guidelines/repository-guidelines/01-new-repository-settings.md:57]: As discussed: please update this paragraph and/or explain when we should create labels using PRs to the `test-infra` repo, and when using the method described in this article.

File: docs/guidelines/repository-guidelines/02-repository-template.md
Comment (klaudiagrz) [inline docs/guidelines/repository-guidelines/02-repository-template.md:6]:
```suggestion
The [`template-repository`](https://github.com/kyma-project/template-repository) offers a unified file, document, and folder structure. Use it for every new repository that you create in Kyma. It helps you to ensure that the project is consistent and standardized.
```

File: docs/guidelines/repository-guidelines/02-repository-template.md
Comment (klaudiagrz) [inline docs/guidelines/repository-guidelines/02-repository-template.md:14]:
```suggestion
* [`.github`](https://github.com/kyma-project/template-repository/tree/main/.github) - This folder contains the pull request template, issue templates, and the Stale Bot that monitors inactive issues, marks them as `stale`, and closes them after the specified period of time.
```

File: docs/guidelines/repository-guidelines/02-repository-template.md
Comment (klaudiagrz) [inline docs/guidelines/repository-guidelines/02-repository-template.md:16]:
```suggestion
* [`docs`](https://github.com/kyma-project/template-repository/tree/main/docs) - In this folder, put the repository-specific documentation only. Store any architectural decisions or documents applicable to all Kyma repositories in the `community` repository.
```

File: docs/guidelines/repository-guidelines/02-repository-template.md
Comment (klaudiagrz) [inline docs/guidelines/repository-guidelines/02-repository-template.md:18]:
```suggestion
* [CODE_OF_CONDUCT.md](https://github.com/kyma-project/template-repository/blob/main/CODE_OF_CONDUCT.md) - This document is a ready-to-use template which provides a link to the general `CODE_OF_CONDUCT.md` document from the `community` repository.
```

File: docs/guidelines/repository-guidelines/02-repository-template.md
Comment (klaudiagrz) [inline docs/guidelines/repository-guidelines/02-repository-template.md:20]:
```suggestion
* [OWNERS](https://github.com/kyma-project/template-repository/blob/main/OWNERS) - In this document, specify the owners of particular parts of your repository. The owners are automatically added as reviewers when you open a pull request that modifies the code and content they own.
```

File: docs/guidelines/repository-guidelines/02-repository-template.md
Comment (klaudiagrz) [inline docs/guidelines/repository-guidelines/02-repository-template.md:22]:
```suggestion
* [OWNERS_ALIASES](https://github.com/kyma-project/template-repository/blob/main/OWNERS_ALIASES) - This file contains the aliases that group the GitHub usernames. Use it when you want to use group names for granting ownership across the repository.
```

File: docs/guidelines/repository-guidelines/02-repository-template.md
Comment (klaudiagrz) [inline docs/guidelines/repository-guidelines/02-repository-template.md:24]:
```suggestion
* [CONTRIBUTING.md](https://github.com/kyma-project/template-repository/blob/main/CONTRIBUTING.md) - This template makes a reference to the [contributing rules](../../contributing/02-contributing.md) that contain the general guidance from the `community` repository and describes the rules for contributing to all Kyma repositories. If there is any additional, project-specific information that you want to add to your project's `CONTRIBUTING.md` document, add them under the same sections as in the general contributing guide.
```

File: docs/guidelines/repository-guidelines/01-new-repository-settings.md
Comment (Ressetkk) [reply docs/guidelines/repository-guidelines/01-new-repository-settings.md:16]: image

File: docs/guidelines/repository-guidelines/01-new-repository-settings.md
Comment (klaudiagrz) [inline docs/guidelines/repository-guidelines/01-new-repository-settings.md:57]: Remove this paragraph, as discussed.

**Review by klaudiagrz — APPROVED** [review]
YAY! 😃

**Pull request merged**
